fix: latent decode bugs in workspaces.go and duplicate json tag in bitbucket.go

## Background

Three small bugs surfaced while reviewing the v0.9.98 (#360) change set. Each is independent and small (< 5 lines), but two of them are already pinned by `BUG`-style tests in `workspaces_test.go`, so the fix needs to flip those tests' expectations rather than just edit the production code.

## Bugs

### 1. `decodeProjects` reads the wrong key for `MaxDepth`

`workspaces.go:224` reads `projectsResponseMap["max_width"]`, but the field is `max_depth` (see `ProjectsRes.MaxDepth` in `workspaces.go:41`). As a result `ProjectsRes.MaxDepth` is always 0 even when the API responds with `max_depth`.

Pinned by `TestDecodeProjects_MaxDepthBug` (`workspaces_test.go:352-370`), whose comment explicitly documents the bug.

Fix: change the key to `max_depth` and invert the test expectation to `int32(5)`.

### 2. `decodeMembers` type-asserts pagination fields against `int` but JSON yields `float64`

`workspaces.go:261-272` reads `page`, `pagelen`, `size` via `.(int)` type assertions on values that come from `encoding/json`, which decodes JSON numbers into `float64` for `interface{}` targets. Every assertion fails, falling through to the `0` default, so `WorkspaceMembers.Page/Pagelen/Size` are always 0 in practice — even when the API returned correct values.

Pinned by `TestDecodeMembers_PaginationFieldsAreZero` (`workspaces_test.go:380-413`).

Fix: assert on `float64` and convert to `int`, mirroring the pattern already used by `decodeWorkspaceList` (`workspaces.go:165-181`). Invert the test expectation.

### 3. `bitbucket.go` duplicate `uuid` json tag

`go vet ./...` on `master` reports:

```
./bitbucket.go:274:2: struct field RefUUID repeats json tag "uuid" also at bitbucket.go:272
```

Two fields in the same struct cannot share a json tag without one of them silently winning (or losing) at unmarshal time. Decide which field actually corresponds to `uuid` in the API response and either rename the other field's tag or set it to `json:"-"`.

## References

- `workspaces.go:224,261-272`, `workspaces_test.go:352-370,380-413`
- `bitbucket.go:272-274`
- v0.9.98: https://github.com/ktrysmt/go-bitbucket/releases/tag/v0.9.98

--- a/bitbucket.go
+++ b/bitbucket.go
@@ -269,9 +269,9 @@ type RepositoryBranchCreationOptions struct {
 type RepositoryBranchDeleteOptions struct {
 	Owner    string `json:"owner"`
 	RepoSlug string `json:"repo_slug"`
-	RepoUUID string `json:"uuid"`
+	RepoUUID string `json:"repo_uuid"`
 	RefName  string `json:"name"`
-	RefUUID  string `json:"uuid"`
+	RefUUID  string `json:"ref_uuid"`
 }
 
 type RepositoryBranchTarget struct {

--- a/workspaces.go
+++ b/workspaces.go
@@ -221,7 +221,7 @@ func decodeProjects(projectResponse interface{}) (*ProjectsRes, error) {
 	if !ok {
 		pagelen = 0
 	}
-	max_depth, ok := projectsResponseMap["max_width"].(float64)
+	max_depth, ok := projectsResponseMap["max_depth"].(float64)
 	if !ok {
 		max_depth = 0
 	}
@@ -258,23 +258,23 @@ func decodeMembers(membersResponse interface{}) (*WorkspaceMembers, error) {
 		members = append(members, *member)
 	}
 
-	page, ok := responseMap["page"].(int)
+	page, ok := responseMap["page"].(float64)
 	if !ok {
 		page = 0
 	}
-	pagelen, ok := responseMap["pagelen"].(int)
+	pagelen, ok := responseMap["pagelen"].(float64)
 	if !ok {
 		pagelen = 0
 	}
-	size, ok := responseMap["size"].(int)
+	size, ok := responseMap["size"].(float64)
 	if !ok {
 		size = 0
 	}
 
 	workspaceMembers := WorkspaceMembers{
-		Page:    page,
-		Pagelen: pagelen,
-		Size:    size,
+		Page:    int(page),
+		Pagelen: int(pagelen),
+		Size:    int(size),
 		Members: members,
 	}
 	return &workspaceMembers, nil

--- a/workspaces_test.go
+++ b/workspaces_test.go
@@ -355,10 +355,7 @@ func TestDecodeProjects_Success(t *testing.T) {
 	assert.Equal(t, "{proj-uuid-1}", result.Items[0].Uuid)
 }
 
-// TestDecodeProjects_MaxDepthBug documents a known bug in decodeProjects:
-// line 214 of workspaces.go reads "max_width" instead of "max_depth",
-// so MaxDepth is always 0 even when "max_depth" is provided in the response.
-func TestDecodeProjects_MaxDepthBug(t *testing.T) {
+func TestDecodeProjects_MaxDepth(t *testing.T) {
 	t.Parallel()
 	response := map[string]interface{}{
 		"page":      float64(1),
@@ -371,8 +368,7 @@ func TestDecodeProjects_MaxDepthBug(t *testing.T) {
 	result, err := decodeProjects(response)
 
 	require.NoError(t, err)
-	// BUG: MaxDepth should be 5 but the code reads "max_width" instead of "max_depth"
-	assert.Equal(t, int32(0), result.MaxDepth, "MaxDepth is always 0 because code reads 'max_width' instead of 'max_depth'")
+	assert.Equal(t, int32(5), result.MaxDepth)
 }
 
 func TestDecodeProjects_InvalidFormat(t *testing.T) {
@@ -383,10 +379,7 @@ func TestDecodeProjects_InvalidFormat(t *testing.T) {
 	assert.Contains(t, err.Error(), "Not a valid format")
 }
 
-// TestDecodeMembers_PaginationFieldsAreZero documents that decodeMembers
-// expects page/pagelen/size as int, but JSON unmarshal produces float64,
-// so pagination fields are always 0 in practice.
-func TestDecodeMembers_PaginationFieldsAreZero(t *testing.T) {
+func TestDecodeMembers_PaginationFields(t *testing.T) {
 	t.Parallel()
 	response := map[string]interface{}{
 		"page":    float64(3),
@@ -411,9 +404,7 @@ func TestDecodeMembers_PaginationFieldsAreZero(t *testing.T) {
 	assert.Equal(t, "testuser", result.Members[0].Username)
 	assert.Equal(t, "Test User", result.Members[0].DisplayName)
 	assert.Equal(t, "{user-uuid}", result.Members[0].Uuid)
-	// BUG: decodeMembers uses type assertion .(int) but JSON produces float64,
-	// so the assertion always fails and pagination fields default to 0.
-	assert.Equal(t, 0, result.Page, "Page is 0 because .(int) type assertion fails on float64")
-	assert.Equal(t, 0, result.Pagelen, "Pagelen is 0 because .(int) type assertion fails on float64")
-	assert.Equal(t, 0, result.Size, "Size is 0 because .(int) type assertion fails on float64")
+	assert.Equal(t, 3, result.Page)
+	assert.Equal(t, 25, result.Pagelen)
+	assert.Equal(t, 1, result.Size)
 }

